tests/everflow/everflow_test_utilities.py

```diff
@@ -618,6 +618,8 @@ def _get_expected_mirror_packet(self, mirror_session, setup, duthost, mirror_pac
         expected_packet.set_do_not_care_scapy(packet.IP, "len")
         expected_packet.set_do_not_care_scapy(packet.IP, "flags")
         expected_packet.set_do_not_care_scapy(packet.IP, "chksum")
+        if duthost.facts["asic_type"] in ["cisco-8000"]:
+            expected_packet.set_do_not_care_scapy(packet.GRE, "seqnum_present")
```

Comment:
Why didn't ignore `GRE.sequence_number`?

Reply:
Hi Bing,
    If you look at  GRE encapsulation for  ERSPAN header for Type-1 & Type-2 shown below.  Only  ERSPAN type-2  has 4 bytes sequence number which is incremented by per session per packet basis. This is  not handled in the current test code so it needs to be ignored while comparing expected packet. For more detail on ERPSAN header reefer to IETF draft:h https://datatracker.ietf.org/doc/html/draft-foschiano-erspan-03

``

         GRE header for ERSPAN type I encapsulation (4 octets [34:37])

       0                   1                   2                   3
       0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
      |0|0|0|0|0|00000|000000000|00000|   Protocol Type for ERSPAN    |
      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+



     GRE header for ERSPAN Type II encapsulation (8 octets [34:41])
       0                   1                   2                   3
       0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
      |0|0|0|1|0|00000|000000000|00000|   Protocol Type for ERSPAN    |
      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
      |      Sequence Number (increments per packet per session)      |
      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+

``

Reply:
@balajib-cisco  Thanks for the clarification. I mean `seqnum_present` is a bit in gre header, while `sequence_number` is the 4 btes you mentioned. Do we intend to ignore the  `seqnum_present` bit only?

Reply:
> @balajib-cisco Thanks for the clarification. I mean `seqnum_present` is a bit in gre header, while `sequence_number` is the 4 btes you mentioned. Do we intend to ignore the `seqnum_present` bit only?
> 
> https://scapy.readthedocs.io/en/latest/api/scapy.layers.l2.html
> 
> field	length	default
> seqnum_present	BitField (1 bit)	0
> sequence_number	XIntField (Cond)	None

Yes. Seqnum_present bit needs to be ignored as it will not be set in everflow type-1 and will be set in type-2. If you look at the test code, except for the outer header 38 bytes( 14 bytes Ether header  + 20 bytes IPv4 header + first 4 bytes of GRE header)  rest of the payload is masked/ignored from the expected packet. This implicitly masking/ignoring GRE header second word containing the sequence number so we do not have to mask it.

Reply:
I see. It makes sense to me. Thanks for the improvement.